### PR TITLE
executor: fix user without process privilege can access cluster_statements_summary_evicted table

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/26125

### What is changed and how it works?

*: What's Changed: fix user without process privilege can access cluster_statements_summary_evicted table

### Check List 

Tests 

- [x] Unit test

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- fix user without process privilege can access cluster_statements_summary_evicted table

### CODE REVIEW

File: infoschema/tables_test.go [hunk 1]
```diff
@@ -1436,6 +1436,16 @@ func (s *testClusterTableSuite) TestStmtSummaryEvictedCountTable(c *C) {
...
+	tk.MustExec("create user 'testuser'@'localhost'")
+	c.Assert(tk.Se.Auth(&auth.UserIdentity{
+		Username: "testuser",
+		Hostname: "localhost",
+	}, nil, nil), Equals, true)
+
+	err := tk.QueryToErr("select * from information_schema.CLUSTER_STATEMENTS_SUMMARY_EVICTED")
+	c.Assert(err, NotNil)
+	// This error is come from cop(TiDB) fetch from rpc server.
+	c.Assert(err.Error(), Equals, "other error: [planner:1227]Access denied; you need (at least one of) the PROCESS privilege(s) for this operation")
```
Comment (mjonss) [inline infoschema/tables_test.go:1448]: Maybe also a test with a new user that only has `Process` privilege?

**Review by mjonss — COMMENTED** [review]
Only minor comment about `PROCESS` vs `Process`. I also tried to add a similar test to `privilege/privileges/privileges_test.go:TestClusterConfigInfoschema` but then it hangs/succeeds when it is suppose to fail, any idea why?
Maybe also add a test for a user with only `Process` privilege.

Otherwise LGTM.

File: executor/infoschema_reader.go [hunk 2]
```diff
@@ -1919,6 +1919,9 @@ func (e *memtableRetriever) dataForTableTiFlashReplica(ctx sessionctx.Context, s
 }
 
 func (e *memtableRetriever) setDataForStatementsSummaryEvicted(ctx sessionctx.Context) error {
+	if e.table.Name.O == infoschema.ClusterTableStatementsSummaryEvicted && !hasPriv(ctx, mysql.ProcessPriv) {
```
Comment (djshow832) [inline executor/infoschema_reader.go:1922]: The same for `statements_summary_evicted`. Please update them together.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by djshow832 — APPROVED** [review]

**Pull request merged**